File: week2/11.Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int maxv = 0;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                maxv = max(maxv,(j-i)*min(height[i],height[j]));
            }
        }
        return maxv;
    }
};

File: week2/11.Container With Most Water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        size = len(height)   
        maxv = 0 
        j = 0
        k = size - 1
        while j < k:
            if height[j] <= height[k]:
                maxv = max(maxv,height[j] * (k - j))
                j += 1
            else:
                maxv = max(maxv,height[k] * (k - j))
                k -= 1
        return maxv
